software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+      set the visible of field "tagLabel" stack "NodeDetails" to false
+      set the visible of image "figureHolder" stack "NodeDetails" to true
+      set the visible of button "loadImgButt" stack "NodeDetails" to true
+      set the visible of button "clearImgButt" stack "NodeDetails" to false 
+      set the visible of button "expandImgButt" stack "NodeDetails" to false 
+      set the visible of button "setCaptureButt" stack "NodeDetails" to true
+      set the visible of button "snapCaptureButt" stack "NodeDetails" to true
+      if tDeets[pLine]["figure"] is not "[empty]" or (tDeets[pLine]["new"] is not empty and tDeets[pLine]["new"] is not "hide") then
+         -- allow clear if 1) [figure] already has a file or 2) a new file has been loaded in [new]
+         -- don't allow clear if "hide" is in [new], which is trying to get rid of what's in [figure]
+         set the visible of button "clearImgButt" stack "NodeDetails" to true 
+         set the visible of button "expandImgButt" stack "NodeDetails" to true 
+      end if
+   else if tLabel is among the words "title memo weight" then
+      set the visible of field "editData" stack "NodeDetails" to true
+      set the visible of field "tagLabel" stack "NodeDetails" to true
+      set the visible of image "figureHolder" stack "NodeDetails" to false
+      set the visible of button "loadImgButt" stack "NodeDetails" to false
+      set the visible of button "clearImgButt" stack "NodeDetails" to false 
+      set the visible of button "expandImgButt" stack "NodeDetails" to false 
+      set the visible of button "setCaptureButt" stack "NodeDetails" to false
+      set the visible of button "snapCaptureButt" stack "NodeDetails" to false
+      put tLabel into field "tagLabel" stack "NodeDetails"
+      if tDeets[pLine]["new"] is not empty and tDeets[pLine]["new"] is not "hide" then
+         put tDeets[pLine]["new"] into field "editData" stack "NodeDetails"
+      else 
+         put tDeets[pLine][tLabel] into field "editData" stack "NodeDetails"
+      end if
+      select the text of field "editData" stack "NodeDetails"
+   end if -- if label is divider
+   set the cNodeDetails of stack "NodeDetails" to tDeets
+end detailsListMouseUp
+
+on loadImgButtMouseUp pLine
+   -- on nodeDetails for loading a figure from an existing file
+   put the cNodeDetails of stack "NodeDetails" into tDeets
+   put specialfolderpath("documents") into tStart
+   answer file "choose an image (it doesn't actually force you to pick an image file at the moment)" with tStart
+   if the result is not cancel then
+      put empty into tDeets[pLine]["new"]
+      put true into tDeets["newFigure"]
+      set the itemdelimiter to slash
+      put it into tImgPath
+      put item -1 of tImgPath into tImgName
+      if sMain[sNowDoc][0]["filepath"] is not empty then
+         put sMain[sNowDoc][0]["filepath"] into tProjFolder
+         delete item -1 of tProjFolder
+         set the defaultfolder to tProjFolder
```

Comment:
@matthewmaier `defaultFolder | folder | directory` is one of those annoying global properties that we really need to ensure we reset to its original setting after each use incase some other code is relying on it. One good way to do this is to have something like the following in a utility library:

````
local sFoldersA

command PushFolder pFolder
    get item 2 of the extents of sFoldersA + 1
    put the folder into sFoldersA[it]
    set the folder to pFolder
end PushFolder

command PopFolder
   get item 2 of the extents of sFoldersA
   if it is not empty then
      set the folder to sFoldersA[it]
      delete variable sFoldersA[it]
   end if
end PopFolder
````